### PR TITLE
Fix for partitioner call method

### PR DESCRIPTION
**Pull request by dmilde**
Solves error "undefined call method" of partitioner.
Working, but I am not a ruby programmer/expert, check needed.

### CODE REVIEW

File: lib/fluent/plugin/kafka_producer_ext.rb [hunk 1]
```diff
@@ -243,14 +245,14 @@ def deliver_messages_with_retries
...
       failed_messages = []
-      partition_count = @cluster.partitions_for(@topic).count
 
       @pending_message_queue.each do |message|
         partition = message.partition
 
         begin
           if partition.nil?
-            partition = Partitioner.call(partition_count, message)
+            partition_count = @cluster.partitions_for(@topic).count
```
Comment (repeatedly) [inline lib/fluent/plugin/kafka_producer_ext.rb:254]: Why move this line?
In `TopicProducer`, `@topic` is fixed value so no need to call for each message.

**Comment by repeatedly** [comment]
Thanks for the patch. I put one comment.

**Comment by dmilde** [comment]
Moved the count back (out of the loop).

File: lib/fluent/plugin/kafka_producer_ext.rb (hunk 1 above)
Comment (dmilde) [reply lib/fluent/plugin/kafka_producer_ext.rb]: Move the count back (out of the loop).

**Comment by repeatedly** [comment]
I confirmed this patch fixes the issue. I will release new version soon. Thanks!

**Pull request merged**